src/Screen/Fields/Button.php

```diff
@@ -124,9 +126,20 @@ public function type(string $visual): self
             self::LIGHT,
             self::DARK,
             self::LINK,
-        ], '', (string) $this->get('class'));
+        ], '', (string)$this->get('class'));
+
+        $this->set('class', $class . ' ' . $visual);
+
+        return $this;
+    }
 
-        $this->set('class', $class.' '.$visual);
+
+    public function loadModalAsync(string $modal, $method, $options, string $text = null): self
```

Comment:
Let's try to keep a single name in the executable file and the template file.
Also the absence of phpdocs makes the whole method look.


Let's write it more strictly, for example, options are declared by default as an array, 
why don't we allow them as `array $options = []`

Reply:
Added phpdoc and wrapper